app/views/devise/sessions/new.html.erb

```diff
@@ -34,7 +35,12 @@
         } %>
       </div>
     <% end -%>
-    <%= f.submit t('account.sign_in'), class: 'buttonGhostM marginTop' %>
+
+    <% if @display_recaptcha %>
+      <%= recaptcha_tags id: 'recaptcha_form', callback: 'recaptcha_verified', expired_callback: 'recaptcha_expired' %>
```

Comment:
`recaptcha_verified` and `recaptcha_expired` are the names of the javascript functions that run when a recaptcha is completed/expires. Recaptchas expire after about 2 minutes.

Reply:
I wonder if this could fit into one of the React components, but I'm still learning React at the moment so I don't have too much opinion on that at the moment. I think @julianguyen might have some better opinions on where this could possibly go?

Reply:
Potentially, but then I'm not sure if we could use the `recaptcha_tags` helper that's provided by the recaptcha gem. Although, I don't think it'd be too hard to recreate what that wrapper does.

Reply:
Thanks for bringing this up and I have a similar sentiment. Yeah, I don't think it's worth the time trying to extract what's going on with the `recapthca_tags` helper so it can be used in React so I think what you have is fine :)